File: December2022/C++/climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
      
        
        vector<int>dp(n+1,0);
   
        int prev1 = 1;
        int prev2 = 1;
        int curr = 1;
        
        for(int i=2;i<=n;i++){
          curr = prev1 + prev2;
            
            prev1 = prev2;
            prev2 = curr;
        }
        
        
        
        return curr;
    }
};

File: December2022/Java/ClimbingStairs.java
// @saorav21994

class Solution {
    public int climbStairs(int n) {
        // fibbonacci
        if (n == 1)
            return 1;
        else if (n == 2) 
            return 2;
        else {
            int res = 0;
            int a = 1, b = 2;
            n -= 2;
            while (n > 0) {
                res = a + b;
                a = b;
                b = res;
                n -= 1;
            }
            return res;
        }
    }
}
